index.js

```diff
@@ -1943,6 +1943,29 @@
         return either.isRight ? [either.value] : [];
       }));
 
+  //# toEither :: a -> b? -> Either a b
+  //.
+  //. Takes a failure value, 'x', and a possibly null or undefined value, 'y'.
+  //. Returns a Right of 'y' if it is defined and not null, otherwise returns
+  //. a Left of 'x'.
+  //.
+  //. See also ['encaseEither'](#encaseEither)
+  //.
+  //. ```javascript
+  //. > S.toEither("No string", "ftp://example.com/")
+  //. Right("ftp://example.com/")
+  //.
+  //. > S.toEither("Invalid protocol", "ftp://example.com/".match(/^https?:\/\//))
+  //. Left("Invalid protocol")
+  //. ```
```

Comment:
So, annoyingly, there's a difference between the results of `match` in node and the browser. In node, the result is an array with some additional properties tacked on. It looks like

```js
[ "https://", index: 0, input "https://example.com" ] // Yes, no `{`!
```

We could use that, though it's a little confusing. I considered dropping `match`, but it's a shame because it's the only standard library function I'm aware of that returns `null` (apart from `querySelector` and its ilk).

I think maybe the very simple example is best?